### PR TITLE
Creating addons section

### PR DESCRIPTION
**Pull request by ndelangen**
This add a section to the guide on how to write addons

### CODE REVIEW

File: content/react/en/creating-addons.md [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+
+You can open a PR on the storybook documentation to have your addon featured there.
+
+## What's next?
+
+This is an example of an addon that uses parameters and displays it in a panel, but that's just one of the many options you have.
+
+With these principles you're able to display you custom UI in a variety of places in the storybook UI.
+
+You can:
```
Comment (zol) [inline content/react/en/creating-addons.md]: Could you please permalink each item to a code example from one of our addons to make it more convenient to find these techniques.

File: content/react/en/creating-addons.md [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+import React, { useMemo } from 'react';
+
+import { useParameter } from '@storybook/api';
+import { addons, types } from '@storybook/addons';
+import { AddonPanel } from '@storybook/components';
+
+// this is often placed in a constants.js file
+const ADDON_ID = 'storybook/parameter';
+const PANEL_ID = `${ADDON_ID}/panel`;
+const PARAM_KEY = `parameter`;
```
Comment (tmeasday) [inline content/react/en/creating-addons.md]: Should this be `assets`

**Comment by domyen** [comment]
@ndelangen @zol is this still active?

**Comment by zol** [comment]
@domyen to the best of my knowledge, this was/is blocked on the 5.2 release

**Comment by ndelangen** [comment]
correct, we need 5.2 to be released, which has the new hooks apis in it.

File: content/react/en/creating-addons.md [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+storiesOf('Component')
+  .addParameters({ assets: ['path/to/asset.png']})
+  .add('variant', () => <Component />);
+```
+
+With our public API defined, let's start writing the addon to make this appear in storybook's UI.
+
+## Writing the addon
+
+Let's create a new file: `./storybook/addons/design-assets.js`. (you may have to create the `addons` folder).
```
Comment (zol) [inline content/react/en/creating-addons.md]: It's not clear which directory is the root here. Is it inside a project that has Storybook installed?

File: content/react/en/creating-addons.md [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+
+The way we'll be attaching the list of assets to stories is via parameters.
+
+This is what it will look like when someone is adding design assets to their story:
+
+```js
+import { storiesOf } from '@storybook/react';
+import Component from './component';
+
+storiesOf('Component')
```
Comment (zol) [inline content/react/en/creating-addons.md]: Could we refactor this guide to use the new CSF?

File: content/react/en/creating-addons.md [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+
+# Creating addons
+
+Storybook's addon architecture allows you to expand it's feature set and customize it to your needs.
+
+Writing your own addon may sound a bit daunting, but as you'll see in this guide, it can be quite simple.
+
+We're going to write an addon together, first let's scope what our addon will do:
+
+## The addon we're going to write
```
Comment (zol) [inline content/react/en/creating-addons.md]: Lets include a screenshot of the finished product to whet people's appetites. I propose a screenshot that includes the taskbox example that we build in LSB with the design-assets addon showing a design that backs one of the stories.

File: content/react/en/creating-addons.md (hunk 4 above)
Comment (ndelangen) [reply content/react/en/creating-addons.md]: ✅

File: content/react/en/creating-addons.md (hunk 3 above)
Comment (ndelangen) [reply content/react/en/creating-addons.md]: ✅

File: content/react/en/creating-addons.md (hunk 5 above)
Comment (ndelangen) [reply content/react/en/creating-addons.md]: I think a bit stylised promo picture would be best, I have no good example of a design asset + story.

To test this addon we have it in the monorepo, but there are no design assets that would make sense to display.

Screenshot 2019-09-23 at 16 08 13

@domyen do you think a picture like the ones here:
https://storybook.js.org/addons/

could be created?

File: content/react/en/creating-addons.md (hunk 1 above)
Comment (ndelangen) [reply content/react/en/creating-addons.md]: ✅

**Comment by ndelangen** [comment]
@zol WDYT, mergeable?

File: content/react/en/creating-addons.md (hunk 5 above)
Comment (zol) [reply content/react/en/creating-addons.md]: @ndelangen The above ss is too disconnected, we want this to make sense to LSB users. Do this:
1. Add the design-assets addon to taskbox
2. Navigate to a good looking story in taskbox
3. Take a ss of the component
4. Link to it in design-assets
5. Take a screenshot (as above) of the design asset sitting next to the story. They should look the same.

**Comment by zol** [comment]
@ndelangen It's almost good to go from my end, just need a better screenshot as per my remaining outstanding comment

File: content/react/en/creating-addons.md (hunk 5 above)
Comment (ndelangen) [reply content/react/en/creating-addons.md]: What's 'taskbox'?

Comment (zol) [reply content/react/en/creating-addons.md]: Taskbox is the name of the app that Learn Storybook teaches people how to build

Comment (ndelangen) [reply content/react/en/creating-addons.md]: 👍

**Comment by ndelangen** [comment]
Screenshot 2019-10-04 at 14 15 18

@zol WDYT?

**Comment by ndelangen** [comment]
Bit more stylised:
Screenshot 2019-10-04 at 14 16 58

**Comment by ndelangen** [comment]
Screenshot 2019-10-04 at 14 56 02

@zol WDYT?

**Comment by zol** [comment]
Looks great @ndelangen . I'm ready to have this merged and shipped!

**Comment by tmeasday** [comment]
@domyen merge when ready!

**Pull request merged**